lang/rust-bootstrap/Portfile

```diff
@@ -278,177 +278,218 @@ livecheck.version           ${version_current}
 # cargo from rust bootstrap is linked against the system's crypto library
```

Comment:
Here’s a command that generates a working dependency list. I think we should include this as a comment in the Portfile:

```sh
# use cargo2port:
# for cl in "${worksrcpath}" "${worksrcpath}/src/bootstrap" "${workpath}/rustc-${version}-src-${build_arch}" "${workpath}/rustc-${version}-src-${build_arch}/src/tools/"*; do test -f "${cl}/Cargo.lock" && cargo2port "${cl}/Cargo.lock"; done | sed '/^cargo\.crates/d' | sed -E 's/([^\\])$/\1 \\/' | sort -u | sed '1s|^|cargo.crates \\\n|' | sed '$s| \\$||' | pbcopy
# cargo2port ${worksrcpath}/../rustc-${version}-src-${build_arch}/src/tools/rust-analyzer/Cargo.lock | pbco
```

Reply:
@essandess I think that proposed command-line is a little long to put on one line, and I think it does have a potential problem that @herbygillot pointed out of missing x86_64 hashes.

I’ve opened an issue on `cargo2port` to accept multiple `Cargo.lock` files on the command-line; the developer doesn’t currently have time to look at it, but if someone wants to take it on (I have no time right now), they would consider a PR. Handling multiple architectures would probably be advisable as well, if that is a need.

Reply:
I do agree that *from my perspective* the update instructions in both Portfiles should be comments *in* the Portfiles, because finding the right Trac ticket is always going to be a bit of a nightmare. This would include an instruction in the rust Portfile that the rust-bootstrap Portfile needs to be updated *before* the rust Portfile for the future.

Reply:
This is the command necessary to generate the dependencies—its length is irrelevant.

The checksums are generated by a separate process. I’d also recommend including these instructions:

```sh
    # set build_arch by hand on arm64/x86_64 systems to get checksums
    # sudo port -d checksum rust rust-bootstrap os.arch=arm build_arch=arm64
    # sudo port -d checksum rust rust-bootstrap os.arch=i386 build_arch=x86_64
    # run `port -f clean --all rust rust-bootstrap` afterwards
```

Reply:
Sorry; I’m not saying that the entire command shouldn't be present. I would present it as a multiline comment, because it’s long and hard to read as one line. Long term, I would want to be able to replace it with a single command for `cargo2port`, which should be able to figure this out (and I think *can* be extended pretty easily to do this).